query-node/mappings/src/content/utils.ts

```diff
@@ -0,0 +1,484 @@
+      assetsProvided: parameters.assets.length,
+      assetIndex: parameters.assetIndex,
+    })
+  }
+
+  // convert asset to data object record
+  return convertAsset({
+    rawAsset: parameters.assets[parameters.assetIndex],
+    db: parameters.db,
+    blockNumber: parameters.blockNumber,
+    contentOwner: parameters.contentOwner,
+  })
+}
+
+/*
+  As a temporary messure to overcome yet-to-be-implemented features in Hydra, we are using redudant information
+  to describe asset state. This function introduces all redudant data needed to be saved to db.
+
+  Changes `result` argument!
+*/
+function integrateAsset<T>(propertyName: string, result: Object, asset: AssetStorageOrUrls) {
+  // helpers - property names
+  const nameUrl = propertyName + 'Urls'
+  const nameDataObject = propertyName + 'DataObject'
+  const nameAvailability = propertyName + 'Availability'
+
+  // is asset saved in storage?
+  if (!isAssetInStorage(asset)) {
+    // (un)set asset's properties
+    result[nameUrl] = asset
+    result[nameAvailability] = AssetAvailability.ACCEPTED
+    result[nameDataObject] = undefined // plan deletion (will have effect when saved to db)
+
+    return result
+  }
+
+  // prepare conversion table between liaison judgment and asset availability
+  const conversionTable = {
+    [LiaisonJudgement.ACCEPTED]: AssetAvailability.ACCEPTED,
+    [LiaisonJudgement.PENDING]: AssetAvailability.PENDING,
+  }
+
+  // (un)set asset's properties
+  result[nameUrl] = [] // plan deletion (will have effect when saved to db)
+  result[nameAvailability] = conversionTable[asset.liaisonJudgement]
+  result[nameDataObject] = asset
+}
+
+async function extractVideoSize(assets: NewAsset[], assetIndex: number | undefined): Promise<BN | undefined> {
+  // escape if no asset is required
+  if (assetIndex === undefined) {
+    return undefined
+  }
+
+  // ensure asset index is valid
+  if (assetIndex > assets.length) {
+    return inconsistentState(`Non-existing asset video size extraction requested`, {assetsProvided: assets.length, assetIndex})
+  }
+
+  const rawAsset = assets[assetIndex]
+
+  // escape if asset is describing URLs (can't get size)
+  if (rawAsset.isUrls) {
+    return undefined
+  }
+
+  // !rawAsset.isUrls && rawAsset.isUpload // asset is in storage
+
+  // extract video size
+  const contentParameters: ContentParameters = rawAsset.asUpload
+  // `size` is masked by `size` special name in struct that's why there needs to be `.get('size') as u64`
+  const videoSize = (contentParameters.get('size') as unknown as u64).toBn()
+
+  return videoSize
+}
+
+async function prepareLanguage(languageIso: string, db: DatabaseManager, blockNumber: number): Promise<Language> {
+  // validate language string
+  const isValidIso = ISO6391.validate(languageIso);
+
+  // ensure language string is valid
+  if (!isValidIso) {
+    return inconsistentState(`Invalid language ISO-639-1 provided`, languageIso)
```

Comment:
I don't think it makes sense to terminate here. There is no way to recover from this. The metadata is not going to change. Instead we should fall back to some default language code or set it to null

Reply:
Invalid state doesn't throw error anymore.